### PR TITLE
validate detector only when start detector; fix flaky test case

### PR DESCRIPTION
**Pull request by ylwu-amzn**
*Issue #, if available:*

## Description of changes:
1. Only validate feature when start detector
2. Fix flaky test case

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/main/java/com/amazon/opendistroforelasticsearch/ad/task/ADTaskManager.java [hunk 1]
```diff
@@ -293,12 +295,6 @@ public void stopDetector(
                 ensureExpectedToken(XContentParser.Token.START_OBJECT, parser.nextToken(), parser);
                 AnomalyDetector detector = AnomalyDetector.parse(parser, response.getId(), response.getVersion());
 
-                String error = validateDetector(detector);
-                if (error != null) {
-                    listener.onFailure(new ElasticsearchStatusException(error, RestStatus.BAD_REQUEST));
-                    return;
-                }
-
```
Comment (ohltyler) [inline src/main/java/com/amazon/opendistroforelasticsearch/ad/task/ADTaskManager.java:301]: Is this where the error was being propagated in the response? If so, just curious what is the reason for trying to stop a non-running task? I guess it makes sense if just for consistency to handle different cases, but just wondering if it's an unnecessary step.

Comment (ylwu-amzn) [reply src/main/java/com/amazon/opendistroforelasticsearch/ad/task/ADTaskManager.java:301]: These code added for start historical detector change. Forgot to clean up it when made stop and delete historical detector change. No need to validate if detector has enabled feature when stop/delete detector.

**Review by ohltyler — APPROVED** [review]
Thanks for the fix!

**Review by saratvemulapalli — APPROVED** [review]
Thanks Yaliang for the change and the IT test to cover this.

**Pull request merged**